### PR TITLE
update to 2023.3.24

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
update dask-ml to 2023.3.24
 - add support for python 3.11
 - scikit-learn >=1.2.0

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -35,8 +34,9 @@ requirements:
     - packaging
     - pandas >=0.24.2
     - python
-    - scikit-learn >=1.0.0
+    - scikit-learn >=1.2.0
     - scipy
+    - importlib-metadata # [py>=310]
```
Comment (ViridianMelody) [inline recipe/meta.yaml]: I don't think `importlib-metadata` is required in run.

**Review by ELundby45 — APPROVED** [review]
Agreed with @ViridianMelody on `importlib-metadata`, I am not seeing that anywhere. Otherwise looks good.

File: recipe/meta.yaml (hunk 1 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml]: yes you are right, I have removed that

**Review by akabanovs — APPROVED** [review]
LGTM

**Pull request merged**